src/mpi/coll/ialltoallv/ialltoallv_tsp_scattered_algos.h

```diff
@@ -93,10 +92,10 @@ int MPIR_TSP_Ialltoallv_sched_intra_scattered(const void *sendbuf, const int sen
     /* Post more send/recv pairs as the previous ones finish */
     for (i = bblock; i < size; i += batch_size) {
         ww = MPL_MIN(size - i, batch_size);
-        index = 0;      /* Store vtcs array from the start */
-        /* Get the send and recv ids from previous sends/recvs.
-         * ((i + j) % bblock) would ensure extracting and storing
-         * dependencies in order from '0, 1,...,bblock' */
+        index = 0;
+        /* Add dependency to ensure not to run until previous block the batch portion
+           is finsihed -- effectively limiting the on-going tasks to bblock 
+        */
```

Comment:
The original comment does not address the real purpose of the code. There is no real dependency here. Dependency is added to effectively limit the on-going jobs.